certora/specs/Account.spec

```
@@ -0,0 +1,488 @@
+            requireInvariant atUniquenessExecutor(
+                require_uint256(_executorPositionOf(module) - 1),
+                require_uint256(_executorPositionOf(otherModule) - 1)
+            );
+            requireInvariant cleanStorageExecutor(require_uint256(_executorLength() - 1));
+        }
+    }
+
+invariant absentValidatorIsNotStored(address module, uint256 index)
+    index < _validatorLength() => (!_validatorContains(module) => _validatorAt(index) != module)
+    filtered { f -> f.selector != sig:execute(bytes32,bytes).selector  && f.selector != sig:executeFromExecutor(bytes32,bytes).selector }
+    {
+        preserved uninstallModule(uint256 moduleTypeId, address otherModule, bytes deInitData) with (env e) {
+            requireInvariant consistencyIndexValidator(index);
+            requireInvariant consistencyKeyValidator(module);
+            requireInvariant atUniquenessValidator(index, require_uint256(_validatorLength() - 1));
+            requireInvariant cleanStorageValidator(require_uint256(_validatorLength() - 1));
+        }
+    }
+
+invariant absentExecutorIsNotStored(address module, uint256 index)
+    index < _executorLength() => (!_executorContains(module) => _executorAt(index) != module)
+    filtered { f -> f.selector != sig:execute(bytes32,bytes).selector  && f.selector != sig:executeFromExecutor(bytes32,bytes).selector }
+    {
+        preserved uninstallModule(uint256 moduleTypeId, address otherModule, bytes deInitData) with (env e) {
+            requireInvariant consistencyIndexExecutor(index);
+            requireInvariant consistencyKeyExecutor(module);
+            requireInvariant atUniquenessExecutor(index, require_uint256(_executorLength() - 1));
+            requireInvariant cleanStorageExecutor(require_uint256(_executorLength() - 1));
+        }
+    }
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│                                                  Module management                                                  │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+// This guarantees that at most one fallback module is active for a given initData (i.e. selector)
+rule fallbackModule(address module, bytes initData) {
+    assert isModuleInstalled(3, module, initData) <=> getFallbackHandler(getDataSelector(initData)) == module;
+}
+
+rule moduleManagementRule(env e, method f, calldataarg args, uint256 moduleTypeId, address module, bytes additionalContext)
+    filtered { f -> !f.isView }
+{
+    bytes context;
+    require context.length == 0;
+
+    bool isEntryPoint = isEntryPoint(e);
+    bool isEntryPointOrSelf = isEntryPointOrSelf(e);
+    bool isExecutionModule = isExecutionModule(e, context);
+
+    bool isModuleInstalledBefore = isModuleInstalled(moduleTypeId, module, additionalContext);
+    f(e, args);
```

Shall we add `@withRevert` here? The following assertion must preserve afaik

```suggestion
    f@withrevert(e, args);
```

---

If the function reverts, than nothing changes, and we are evaluating  a "non-transition".

IMO we don't need it. Adding it should passs, but it would also make the system consider more cases, so maybe the proving will be slower ?